src/ln/channel.rs

```diff
@@ -2012,7 +2012,11 @@ impl Channel {
 		if self.channel_outbound && msg.scriptpubkey.len() > 34 {
 			return Err(HandleError{err: "Got shutdown_scriptpubkey of absurd length from remote peer", action: None});
 		}
-		//TODO: Check shutdown_scriptpubkey form as BOLT says we must? WHYYY
+
+		//Check shutdown_scriptpubkey form as BOLT says we must
+		if !(msg.scriptpubkey.is_p2pkh())&&!(msg.scriptpubkey.is_p2sh())&&!(msg.scriptpubkey.is_v0_p2wpkh())&&!(msg.scriptpubkey.is_v0_p2wsh()){
```

Comment:
nit: adding a space before/after AND operator isn't nicer for code readability ?